base/essentials.jl

```diff
@@ -643,19 +643,19 @@ function append_any(xs...)
             end
             for j in 1:lx
                 y = @inbounds x[j]
-                arrayset(true, out, y, i)
+                arrayset(false, out, y, i)
                 i += 1
             end
-        elseif x isa Tuple
-            lx = length(x)
+        elseif x isa Tuple || x isa NamedTuple
```

Comment:
this'll break performance: we want the optimizer to copy the below code block so it can optimize it, so you'll need to copy it manually here